src/main/java/mediator/ConcreteMediator.java

```diff
@@ -0,0 +1,287 @@
+                        carriers.add(carrier);
+                        ru.setupCarrier(carrier);
+                        return;
+                    }
+                    default -> {
+                        System.out.printf(
+                                "[ERROR] Invalid number of RF ports. Requires 2 or 4, but got %s%n", rfPorts.size());
+                        return;
+                    }
+                }
+            }
+        }
+        System.out.printf("[ERROR] No RU with that name %s is registered with the system%n", name);
+    }
+
+    /**
+     * Creates a carrier that will then be registered with the system. This carrier does not need to be added
+     * to an RU upon creation, but will instead be available to be added to an RU at a later time.
+     * Note: The demo 1 implementation does not make use of this functionality.
+     *
+     * @param rfPorts A list of RF ports that will be used for this carrier.
+     * @param carrierFrequencies The frequency band that will be used with this carrier.
+     * @param transmittingPower The transmitting power for this carrier.
+     */
+    @Override
+    public synchronized void createCarrier(List<RfPort> rfPorts, FrequencyBand carrierFrequencies, Double transmittingPower) {
+        switch (rfPorts.size()) {
+            case 2 -> carriers.add(carrierManagement.createWcdmaCarrier(rfPorts, carrierFrequencies, transmittingPower));
+            case 4 -> carriers.add(carrierManagement.createLteCarrier(rfPorts, carrierFrequencies, transmittingPower));
+            default -> {
+                System.out.printf(
+                        "[ERROR] Invalid number of RF ports. Requires 2 or 4, but got %s%n", rfPorts.size());
+                return;
+            }
+        }
+    }
+
+    /**
+     * Displays the carriers associated with an RU.
+     *
+     * @param name The name of the RU for which the carriers will be displayed.
+     */
+    @Override
+    public void displayCarrierOnRu(String name) {
+        AtomicBoolean flag = new AtomicBoolean(false);
+        radioUnits.forEach(ru -> {
+            if (ru.getRadioUnitName().equals(name)) {
+                flag.set(true);
+                if (ru.getCarriers().size() == 0) {
+                    System.out.printf(
+                            "[ERROR] The RU with the name %s has no carriers associated with it.%n", name);
+                } else {
+                    ru.getCarriers().forEach(carrier -> System.out.println(carrier.toString()));
+                }
+                return;
+            }
+        });
+        if (!flag.get()) {
+            System.out.printf(
+                    "[ERROR] No RUs with the name %s have been registered with the system.%n", name);
+        }
+    }
+
+    /**
+     * Prints all Carriers that have been created within the system. These do
+     * not need to be carriers associated with an RU.
+     */
+    @Override
+    public void printCreatedCarriers() {

```

Comment:
I'm not entirely sure which requirement this is satisfying - my understanding is that a carrier must always be associated with an RU (from a real life point of view, they don't exist; from a requirements point of view, why else would they have an RF port?), so holding a second list of carriers here might not be necessary.
Printing all carriers could then be achieved by iterating over all the RUs, and listing the carriers setup on those.

Reply:
So I was iffy on this as well. But I think I must have just misunderstood 1.8.3 while I was reviewing it over the weekend because I am in agreement, a carrier will only ever be created and then immediately associated with an RU. There's no need to have it in a limbo state where it could be associated with an RU after the fact (at least according to what Ista has written). So I'll revert this and remove the carrier list since it really won't be needed.